modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -117,75 +140,180 @@ template<typename Net> class Params {
+    reshaped based on this information and your inputs will be preprocessed for
+    new input sizes. Follow https://docs.openvinotoolkit.org/latest/classInferenceEngine_1_1networkNetwork.html
+    for additional information.
+
+    @param reshape_table Map of pairs: name of corresponding data and its dimension.
+    @return the reference on modified object.
+    */
     Params<Net>& cfgInputReshape(const std::map<std::string, std::vector<std::size_t>>& reshape_table) {
         desc.reshape_table = reshape_table;
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::string&& layer_name, std::vector<size_t>&& layer_dims) {
-        desc.reshape_table.emplace(layer_name, layer_dims);
+    /** @overload
+    Function with a rvalue parameters.
+
+    @param reshape_table rvalue map of pairs: name of corresponding data and its
+    dimension.
+    @return the reference on modified object.
+    */
+    Params<Net>& cfgInputReshape(std::map<std::string, std::vector<std::size_t>>&& reshape_table) {
+        desc.reshape_table = std::move(reshape_table);
         return *this;
     }
 
+    /** @overload
+    Function for one input layer.
+
+    @param layer_name Name of layer.
+    @param layer_dims New dimensions for this layer.
+    @return the reference on modified object.
+    */
     Params<Net>& cfgInputReshape(const std::string& layer_name, const std::vector<size_t>& layer_dims) {
         desc.reshape_table.emplace(layer_name, layer_dims);
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::unordered_set<std::string>&& layer_names) {
-        desc.layer_names_to_reshape = std::move(layer_names);
+    /** @overload
+    Function with a rvalue parameters for one input layer.
+
+    @param layer_name rvalue name of layer.
+    @param layer_dims rvalue new dimensions for this layer.
+    @return the reference on modified object.
+    */
+    Params<Net>& cfgInputReshape(std::string&& layer_name, std::vector<size_t>&& layer_dims) {
+        desc.reshape_table.emplace(layer_name, layer_dims);
         return *this;
     }
 
+    /** @overload
+    Function for reshape by image size.
+
+    The function is used to set names of layers that will be used for network reshape.
+    Dimensions will be constructed automatically by current network input and height and
+    width of image.
+
+    @param layer_names set of names of network layers that will be used for network reshape.
+    @return the reference on modified object.
+    */
     Params<Net>& cfgInputReshape(const std::unordered_set<std::string>& layer_names) {
         desc.layer_names_to_reshape = layer_names;
         return *this;
     }
 
+    /** @overload
+    Function with a rvalue parameters for reshape by image size.
+
+    @param layer_names rvalue set of the selected layers will be reshaped automatically
+    its input image size.
+    @return the reference on modified object.
```

Comment:
No need to document the rvalue specifics, simply mention the overload.

Reply:
@TolyaTalamanov  ?

Reply:
Done